Requests to Drupal.org update feed can fail with gnarly error

If HTTP requests fail in generators/lib/drupalProjectVersion.js for latestRelease() or latestReleaseStable(), an error is returned 'No version data found.' At least one caller in the codebase is not properly handling that error, so the system is crashing with that as a vague clue:

```
✖ No version data found.
events.js:165
      throw err;
      ^

Error: Uncaught, unspecified "error" event. (No version data found.)
    at child.emit (events.js:163:17)
    at Immediate.<anonymous> (/usr/local/lib/node_modules/generator-outrigger-drupal/node_modules/yeoman-generator/lib/index.js:424:16)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
[ERROR] Error running generator outrigger/generator : 1
```

Fix the error handling in the (currently unidentified) caller.

--- a/generators/app/index.js
+++ b/generators/app/index.js
@@ -65,7 +65,7 @@ module.exports = Generator.extend({
         var done = this.async();
         options.drupalDistro.releaseVersion(options.drupalDistroVersion, done, function(err, version, done) {
           if (err) {
-            this.log.error(err);
+            this.env.error('Could not retrieve distribution project info: ' + err);
             return done(err);
           }
           options.drupalDistroRelease = version;
@@ -86,7 +86,7 @@ module.exports = Generator.extend({
           drupalOrgApi.latestRelease(options.cacheInternal, options.drupalDistroVersion, done,
             function(err, version, done) {
               if (err) {
-                this.log.error(err);
+                this.env.error('Could not retrieve project info for selected cache system: ' + err);
                 return done(err);
               }
               options.cacheVersion = version.substr(4);
@@ -107,7 +107,7 @@ module.exports = Generator.extend({
           drupalOrgApi.latestRelease('smtp', options.drupalDistroVersion, done,
             function(err, version, done) {
               if (err) {
-                this.log.error(err);
+                this.env.error('Could not retrieve project info for selected mail system: ' + err);
                 return done(err);
               }
               options.smtpVersion = version.substr(4);

--- a/generators/lib/drupalProjectVersion.js
+++ b/generators/lib/drupalProjectVersion.js
@@ -66,7 +66,7 @@ function init() {
         return cb(null, result.releases[0].release[0].version[0], done);
       }
       else {
-        err = 'No version data found.';
+        err = 'No version data found. This may be a an error in communication with updates.drupal.org.';
       }
       if (err) {
         return cb(err, null, done);
@@ -86,7 +86,7 @@ function init() {
         return cb(null, release.version[0], done);
       }
       else {
-        err = 'No version data found.';
+        err = 'No version data found. This may be a an error in communication with updates.drupal.org.';
       }
 
       if (err) {

